duckduckgo/Android
# Feature/webview improvements

Asana:
https://app.asana.com/0/471012189846020/481692274328411

**Description**:
Improve the browsing experience by making the following changes
1. Remove `NestedScrollView` as a wrapper around the `WebView` as it completely messed with the browser experience.
1. Introducing a `WebView` subclass which allows the toolbar to still react to swipes. This is basically a straight Kotlin port of the java implementation provided here https://github.com/takahirom/webview-in-coordinatorlayout. It's not perfect, but might be a good base to build upon. The code in here could do with tidying; as I say, a straight port of that repo's version. 
1. Instructing Android not to destroy the `BrowserActivity` on common config changes. The Activity getting destroyed was the main reason why the `WebView` didn't handle rotation etc... well
1. Better support pinch to zoom by hiding the vile default zoom controls that appear
1. Removing swipe to refresh on the web view. Would be nice to do this, but it doesn't play nice with the nested web view approach, so will require a little more work to implement. Will be handled here https://app.asana.com/0/414730916066338/483384580258310

**Steps to test this PR**:
1. Load page
1. Rotate
    - Ensure the WebView doesn't completely reload the web page and resizes appropriately
1. Scroll page up and down
    - Ensure the toolbar hides and reappears as expected
1. Pinch to zoom
    - Ensure the web page zooms in/out accordingly
